subxt/src/client/online_client.rs

```diff
@@ -131,14 +134,34 @@ impl<T: Config> OnlineClient<T> {
         {
             // Try to fetch the latest unstable metadata, if that fails fall back to
             // fetching the latest stable metadata.
-            const V15_METADATA_VERSION: u32 = u32::MAX;
-            match rpc.metadata_at_version(V15_METADATA_VERSION).await {
+            match rpc.metadata_at_version(UNSTABLE_METADATA_VERSION).await {
                 Ok(bytes) => Ok(bytes),
-                Err(_) => rpc.metadata().await,
+                Err(_) => OnlineClient::fetch_latest_stable_metadata(rpc).await,
             }
         }
 
         #[cfg(not(feature = "unstable-metadata"))]
+        OnlineClient::fetch_latest_stable_metadata(rpc).await
+    }
+
+    /// Fetch the latest stable metadata from the node.
+    async fn fetch_latest_stable_metadata(rpc: &Rpc<T>) -> Result<Metadata, Error> {
```

Comment:
I think, since Subxt supports V14 and V15 metadata explicitly and doesn't support others (atm), I'd be explicit about trying to get V15 metadata from the call (incase some node decides to return some other version that we don't support decoding from or whatever).

I pondered removing the `unstable-metadata` feature flag but I think I'm happy keeping it as an "opt-in that will fall back to latest supported stable if not exists", just because why not. :)

So let's explicitly ask for V15 here, and if that fails, call `rpc.metadata().await` to get V14 which we also know we can decode (little easier anyway since we don't need to bother querying versions even if we don't want to). Then we'll keep working regardless of any future/different metadata versions, and can upgrade subxt to support them whenever we like.